src/Angor/Client/Components/MessageComponent.razor

```diff
@@ -0,0 +1,444 @@
+                                    </div>
+                                }
+                                else
+                                {
+                                    <div class="messages-list p-2">
+                                        @foreach (var message in MessageService.DirectMessages.Where(m => m != null))
+                                        {
+                                            <div class="message-item @(message.IsFromCurrentUser ? "outgoing" : "incoming")">
+                                                <div class="message-bubble">
+                                                    <div class="message-content">@message.Content</div>
+                                                    <div class="message-timestamp">@message.Timestamp.ToLocalTime().ToString("HH:mm")</div>
+                                                </div>
+                                            </div>
+                                        }
+                                    </div>
+                                }
+                            </div>
+
+                            <!-- Input -->
+                            <div class="chat-input mt-3">
+                                <div class="input-group">
+                                    <input type="text" class="form-control"
+                                           placeholder="Type a message..."
+                                           @bind="newMessage"
+                                           @onkeypress="@(async e => { if(e.Key is "Enter" or "NumpadEnter") await SendMessageAsync(); })"
+                                           disabled="@MessageService.IsSendingMessage" />
+                                    <button class="btn btn-border-success"
+                                            @onclick="SendMessageAsync"
+                                            disabled="@MessageService.IsSendingMessage">
+                                        @if (MessageService.IsSendingMessage)
+                                        {
+                                            <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>
+                                        }
+                                        else
+                                        {
+                                            <Icon IconName="send" Height="20" Width="20" />
+                                        }
+                                    </button>
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+    </div>
+</div>
+
+@code {
+    [Parameter] public string ContactNpub { get; set; }
+    [Parameter] public string ContactHexPub { get; set; }
+    [Parameter] public string CurrentUserNpub { get; set; }
+    [Parameter] public string CurrentUserPrivateKeyHex { get; set; }
```

Comment:
If you get the private key the CurrentUserNpub should be derived from it

Reply:
ah yeah good point

Reply:
Actually it is derived
```
            var words = await passwordComponent.GetWalletAsync();
            var founderNostrPrivateKey = DerivationOperations.DeriveProjectNostrPrivateKey(words, FounderProject.ProjectInfo.FounderKey);
            currentUserPrivateKeyHex = Encoders.Hex.EncodeData(founderNostrPrivateKey.ToBytes());
```